### ISSUE TITLE
Write 3D TIFF files from volumes

### ISSUE THREAD

**Issue by cryos** [OpenChemistry/tomviz#39]
VTK doesn't currently support writing 3D TIFFs, add support and expose this in tomviz.

**Comment by cryos** [comment]
You wouldn't believe how convoluted this was, I am close to having a working patch but have largely rewritten the TIFF writer, and still need to clean up the patch. The TIFF library isn't easy to debug, and the documentation for their API was quite terse... Sorry it took longer than expected.

**Comment by cryos** [comment]
To clarify - I have a working patch, but need to clean it up. I was able to round trip data.

**Comment by cryos** [comment]
Waiting on build bots, and some review, but the patch is there with a merge request for VTK. Next I need to push ParaView forward, and make a merge request with a modified writer XML, and then we can push tomviz' ParaView forward. The VTK merge request for your reference,

https://gitlab.kitware.com/vtk/vtk/merge_requests/298

**Comment by cryos** [comment]
Forgot to close this, it was merged and should be present in the 0.5 binaries. Testing and verification encouraged as we work towards our next release.

**Comment by Hovden** [comment]
I can't round trip the data.

It loads with an incorrect extent: it appears to wrap half the values to negative. The Nanoparticle Tiltseries data packaged with the software has all positive values. But when it is saved with the tiff writer it takes on negative values (shown below)
![screen shot 2015-06-23 at 10 59 27 pm](https://cloud.githubusercontent.com/assets/6197997/8321960/f372d766-19fb-11e5-9406-60e618d524fa.png)

**Comment by cryos** [comment]
I will take a look now, not what I saw in my testing.

**Comment by cryos** [comment]
Tracked the source of the issue, I had tested on signed volumes and we are not correctly tagging the volumes as unsigned when necessary. I will get a patch in soon, it may take a few days to filter up to ParaView. Thanks for reporting the problem.

**Comment by cryos** [comment]
This is awaiting merge into VTK, and then I will see about moving the ParaView version forward and getting fresh binaries. I will update this report once the relevant pieces have been moved forward.

**Comment by cryos** [comment]
Pushed it forward, binaries will be available soon, verified on char and short volumes in addition to IEEE float that already worked well.
